iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -84,132 +74,341 @@ public double getA() {
         return a;
     }
 
-    private Complex computeU2(Complex y1, Complex y2, Complex z, double rho) {
-        Complex v1p = v1.multiply(rho); // v1p=v1*rho
-        Complex i1p = i1.divide(rho); // i1p=i1/rho
-
-        Complex i2p = i1p.subtract(y1.multiply(v1p)); // i2p=i1p-y1*v1p
-        Complex v2 = v1p.subtract(z.multiply(i2p)); // v2p=v1p-z*i2
-        Complex i2 = i2p.subtract(y2.multiply(v2)); // i2=i2p-y2*v2
-        Complex s2 = v2.multiply(3).multiply(i2.conjugate()); // s2=3*v2*conj(i2)
-        return v2.multiply(Math.sqrt(3f));
+    public int getEnd() {
+        return end;
     }
 
-    private Complex computeU2(Complex y, Complex z, double rho) {
-        return computeU2(y, new Complex(0, 0), z, rho);
+    /**
+     * @deprecated Not used anymore. Use
+     * {@link SV#otherSide(double, double, double, double, double, double, double)}
+     */
+    @Deprecated
+    public SV otherSide(double r, double x, double g, double b, double rho) {
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
-    private Complex computeS2(Complex y1, Complex y2, Complex z, double rho) {
-        Complex v1p = v1.multiply(rho); // v1p=v1*rho
-        Complex i1p = i1.divide(rho); // i1p=i1/rho
-
-        Complex i2p = i1p.subtract(y1.multiply(v1p)); // i2p=i1p-y1*v1p
-        Complex v2 = v1p.subtract(z.multiply(i2p)); // v2p=v1p-z*i2
-        Complex i2 = i2p.subtract(y2.multiply(v2)); // i2=i2p-y2*v2
-        return v2.multiply(3).multiply(i2.conjugate()); // s2=3*v2*conj(i2)
+    public SV otherSide(double r, double x, double g1, double b1, double g2, double b2, double rho) {
+        return otherSide(r, x, g1, b1, g2, b2, rho, 0.0);
     }
 
-    private Complex computeS2(Complex y, Complex z, double rho) {
-        return computeS2(y, new Complex(0, 0), z, rho);
+    public SV otherSide(double r, double x, double g1, double b1, double g2, double b2, double rho, double alpha) {
+        LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1 / rho, -alpha, 1.0, 0.0,
+            new Complex(g1, b1), new Complex(g2, b2));
+        return otherSide(adm);
     }
 
-    public SV otherSide(double r, double x, double g, double b, double rho) {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y = new Complex(g, b); // y=g+jb
-        Complex u2 = computeU2(y, z, rho);
-        Complex s2 = computeS2(y, z, rho);
-        return new SV(-s2.getReal(), -s2.getImaginary(), u2.abs(), Math.toDegrees(u2.getArgument()));
+    // Has been modified
```

Comment:
No need to comment on the previous implementation

Reply:
Done